src/modules/animate/index.tsx

```diff
@@ -0,0 +1,584 @@
+import { FFmpeg } from '@ffmpeg/ffmpeg';
+import { fetchFile, toBlobURL } from '@ffmpeg/util';
+import { Button, Group, NumberInput, Popover, Progress, Slider, Stack, Text, createStyles } from '@mantine/core';
+import { fabric } from 'fabric';
+import React, { useEffect, useRef, useState } from 'react';
+import SectionTitle from '../../components/SectionTitle';
+import { FABRIC_JSON_ALLOWED_KEYS } from '../../constants';
+import { generateId } from '../../utils';
+import { interpolatePropertyValue } from './helpers';
+import { addVideoToCanvas } from '../image/helpers';
+interface AnimationProps {
+	canvas: fabric.Canvas;
+	currentSelectedElements: fabric.Object[];
+	saveArtboardChanges: () => void;
+}
+
+export interface Keyframe {
+	id: string;
+	property: string;
+	value: string | number;
+	timeMark: number;
+	easing: string;
+}
+
+const MAX_DURATION = 60 * 1000;
+
+const Animation: React.FC<AnimationProps> = ({ currentSelectedElements, saveArtboardChanges, canvas }) => {
+	const { classes } = useStyles();
+	const [keyframes, setKeyframes] = useState<Keyframe[]>([]);
+	const [hasChanges, setHasChanges] = useState<boolean>(false);
+	const [duration, setDuration] = useState<number>(3000);
+	const [fps, setFps] = useState<number>(30);
+	const [isPlaying, setIsPlaying] = useState<boolean>(false);
+	const [timemark, setTimemark] = useState<number>(0);
+	const [loaded, setLoaded] = useState(false);
+	const [progress, setProgress] = useState({
+		percent: 0,
+		message: '',
+		rendering: false,
+	});
+	const artboardRef = useRef<fabric.Rect | null>(null);
+
+	const ffmpegRef = useRef(new FFmpeg());
+
+	const loadFfmpeg = async () => {
+		setLoaded(false);
+		setProgress({
+			percent: 0,
+			message: 'Initialising encoder',
+			rendering: true,
+		});
+		const baseURL = 'https://unpkg.com/@ffmpeg/core-mt@0.12.5/dist/esm';
```

Comment:
i know including these lazyly as npm module will be requiring lots of config, lets revisit this when we get time.